README.md

```diff
@@ -1,8 +1,130 @@
+const root = document.querySelector('[role=application]')
+const data = root.dataset
+
+const client = new CozyClient({
+  link: new CozyStackLink({
+    uri: `${window.location.protocol}//${data.cozyDomain}`,
+    token: data.cozyToken
+  })
+})
+```
+
+#### Creating a provider
+
+To connect the client to your component tree, use a `CozyProvider` somewhere high in your app:
+
+```js
+import CozyClient, { CozyProvider } from 'cozy-client'
+
+const client = new CozyClient({
+  /*...*/
+})
+
+ReactDOM.render(
+  <CozyProvider client={client}>
+    <MyApp />
+  </CozyProvider>,
+  document.getElementById('main')
+)
+```
+
+#### Integrating with an existing redux store
+
+`cozy-client` uses redux internally to centralize the statuses of the various fetches and replications triggered by the library, and to store locally the data in a normalized way. If you already have a redux store in your app, you can configure `cozy-client` to use this existing store:
+
+```js
+import CozyClient from 'cozy-client'
+import { combineReducers, createStore, applyMiddleware } from 'redux'
+import myReducers from './myapp/reducers'
+import myMiddleware from './myapp/middleware'
+
+const client = new CozyClient({
+  /*...*/
+})
+
+const store = createStore(
+  combineReducers({ ...myReducers, cozy: client.reducer() }),
+  applyMiddleware(myMiddleware)
+)
+```
+
+### Requesting data
+
+To make it easy to fetch data and make it available to your component, we provide a higher-order component called `connect`. Basic example of usage:
+
+```jsx
+import React from 'react'
+import { connect, find } from 'cozy-client'
+
+const query = find('io.cozy.todos').where({ checked: false })
+
+const TodoList = ({ data, fetchStatus }) =>
+  fetchStatus !== 'loaded'
+    ? <h1>Loading...</h1>
+    : <ul>{data.map(todo => <li>{todo.label}</li>)}</ul>
+
+export default connect(query)(TodoList)
+```
+
+When we use `connect` to bind a query to a component, three things happen:
+ - The query passed as an argument will be executed when the component mounts, resulting in the loading of data from the client-side store, or the server if the data is not in the store ;
+ - Our component subscribes to the store, so that it is updated if the data changes ;
+ - props are injected into the component: if we were to declare `propTypes` they would look like this:
+```jsx
+TodoList.propTypes = {
+  fetchStatus: PropTypes.string.isRequired,
+  data: PropTypes.array
+}
+```
+
+#### The injected props
+
+As seen above, `connect` will pass the result of the query fetch to the wrapped component as a set of props. For list fetches, the injected props are the following:
+ - `data`: an array of documents
+ - `fetchStatus`: the status of the fetch (`pending`, `loading`, `loaded` or `error`)
+ - `lastFetch`: when the last fetch occured
+ - `hasMore`: the fetches being paginated, this property indicates if there are more documents to load
+
+#### The query Domain-Specific Language (DSL)
```

Comment:
I would talk about _« The query API »_. For me, and maybe I am wrong, a DSL is some words that define the domain. For Cozy-Stack, it's words like _"doctype"_, _"permissions"_, etc.

Reply:
Effectivement, on devrait peut-être plus parler d'implémentation de DSL, mais le raccourci est utilisé un peu partout, je ne pense pas que ça choque grand monde. Mais :+1: pour utiliser "Making Queries" (Django = :heart:). En tout cas, un DSL n'est pas forcément compliqué (avec un lexer...), il existe plein de patterns et celui-ci en l'occurence est un [Expression Builder](https://martinfowler.com/dslCatalog/expressionBuilder.html).